code/game/gamemodes/miniantags/abduction/abduction_surgery.dm

```diff
@@ -1,100 +1,83 @@
-/datum/surgery/organ_extraction
-	name = "Experimental Dissection"
-	steps = list(/datum/surgery_step/generic/cut_open, /datum/surgery_step/generic/clamp_bleeders, /datum/surgery_step/generic/retract_skin, /datum/surgery_step/open_encased/saw, /datum/surgery_step/open_encased/retract, /datum/surgery_step/internal/extract_organ, /datum/surgery_step/internal/gland_insert, /datum/surgery_step/generic/cauterize)
-	possible_locs = list("chest")
+/datum/surgery_step/internal/manipulate_organs/abduct
+	priority = 20 //DO THIS NOT THE IMPLANT ORGAN ONE!
+	possible_locs = list("chest", "head")
+	time = 32
+	requires_organic_bodypart = FALSE // Also for IPCs
 
-/datum/surgery/organ_extraction/can_start(mob/user, mob/living/carbon/target, target_zone, obj/item/tool,datum/surgery/surgery)
-	if(!ishuman(user))
-		return FALSE
-	if(ishuman(target))
-		var/mob/living/carbon/human/H = target
-		var/obj/item/organ/external/affected = H.get_organ(target_zone)
-		if(!affected)
-			return FALSE
-		if(affected.is_robotic())
-			return FALSE
-	var/mob/living/carbon/human/H = user
-	// You must either: Be of the abductor species, or contain an abductor implant
-	if((isabductor(H) || (locate(/obj/item/implant/abductor) in H)))
-		return TRUE
-	return FALSE
+/datum/surgery_step/internal/manipulate_organs/abduct/is_valid_target(mob/living/carbon/human/target)
+	return ishuman(target)
+
+/datum/surgery_step/internal/manipulate_organs/abduct/is_valid_user(mob/living/carbon/human/user)
+	return ishuman(user) && (isabductor(user) || (locate(/obj/item/implant/abductor) in user))
 
-/datum/surgery_step/internal/extract_organ
+/datum/surgery_step/internal/manipulate_organs/abduct/extract_organ
 	name = "remove heart"
-	accept_hand = 1
-	time = 32
-	var/obj/item/organ/internal/IC = null
+	accept_hand = TRUE
+	surgery_start_stage = list(SURGERY_STAGE_OPEN_INCISION, SURGERY_STAGE_OPEN_INCISION_BONES, SURGERY_STAGE_ROBOTIC_HATCH_OPEN)
+	next_surgery_stage = SURGERY_STAGE_SAME
+	allowed_surgery_tools = SURGERY_TOOLS_EXTRACT_ORGAN
+	var/obj/item/organ/internal/heart/foundHeart = null
 
-/datum/surgery_step/internal/extract_organ/begin_step(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
-	for(var/obj/item/I in target.internal_organs)
+/datum/surgery_step/internal/manipulate_organs/abduct/extract_organ/begin_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	for(var/obj/item/I in target.get_organs_zone(target_zone))
 		// Allows for multiple subtypes of heart.
 		if(istype(I, /obj/item/organ/internal/heart))
-			IC = I
+			foundHeart = I
 			break
-	user.visible_message("[user] starts to remove [target]'s organs.", "<span class='notice'>You start to remove [target]'s organs...</span>")
+	if(!foundHeart)
+		to_chat(user, "<span class='warning'>You can't seem to find an organ that functions like a heart in [target].</span>")
+		return -1
```

Comment:
I think it would be better to use a define in these return values instead of a magic number

Reply:
Will change it. There are some inconsistencies with this as well so it's needed.